### PR TITLE
feat: add leverage/Params.max_share_trigger proto

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

We need a leverage module parameter to solve a cold start / bootstrap problem for tokens with limited `max_supply` and `max_collateral_share`.

Ref: 
* https://github.com/umee-network/umee/issues/1227#issuecomment-1233167479
* https://www.notion.so/Initial-token-settings-for-x-leverage-afcde23e5b2f404bb957ef00b09cb5ac

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/design_docs/010-market-params.md [hunk 1]
```diff
@@ -156,8 +158,10 @@ Let's draw the following scenario to picture the liquidators risk:
...
 ### Positive
+
 - Multiple ways of preserving market health
 - Per-token parameters allow fine grained control
 
 ### Negative
-- Users may find restrictions unfair, e.g. "why can't I borrow just because other people's collateral is too high?"
\ No newline at end of file
+
+- Users may find restrictions unfair, e.g. "why can't I borrow just because other people's collateral is too high?"
```
Comment (robert-zaremba) [inline docs/design_docs/010-market-params.md:167]: auto formatting

**Comment by toteki** [comment]
This should wait until v4, as it is not required for mainnet release.

**Comment by robert-zaremba** [comment]
@adamewozniak , @toteki I prefer to have stable proto. This PR doesn't add new functionality. But we will need to use it as soon as we will enable Axelar.
Also, see: https://umeeworkspace.slack.com/archives/C03CVS0459A/p1663499718056569

**Comment by robert-zaremba** [comment]
I'm going to extract the docs imprvements from this PR and move it to a new PR.

**Pull request closed without merging**